src/internal/internal_getrf_tntpiv.cc

```diff
@@ -329,8 +329,12 @@ void getrf_tntpiv_local(
     blas::device_memcpy<device_pivot_int>( &hipiv[0], dipiv, diag_len,
                                    blas::MemcpyKind::Default, *queue);
 
-    device_copy_vector( diag_len, dA, mlocal + 1,
-                                  &hdiagu[ 0 ], 1, *queue );
+    device_info_int host_info;
+    blas::device_memcpy<device_info_int>( &host_info, dinfo, 1, *queue );
```

Comment:
If we fuse this with the preceding memcpy, we avoid adding an extra round-trip latency.  (One test I ran on Frontier a month or two ago seemed to put the panel on the critical path.)
The data is already sequential in device memory, and we can `static_assert(sizeof(device_info_int) <= sizeof(device_pivot_int))` which Lapack++ currently always provides.

Reply:
Would we need `sizeof( device_info_int ) == sizeof( device_pivot_int )` for
```
blas::device_memcpy<device_pivot_int>( ..., diag_len + 1, ... );
```
to work? cuSolver can have `int64` pivots and `int` info, which is really odd since they both contain row indices. In that case, copying `diag_len + 1` 64-bit elements would overflow the buffer by 32 bits.

Instead, we could copy as bytes:
```
blas::device_memcpy<char>( ..., sizeof( device_pivot_int )*diag_len + device_info_int, ... );
```
In that case, we don't care about their relative sizes.

Reply:
I guess I assumed it'd then allocate `sizeof(device_pivot_int)` bytes for the info.  But, I agree that it's probably not worth bothering with right now.

Reply:
You're right, if the buffer has (diag_len + 1) 64-bit entries on both host and device, then overflow isn't an issue. Just need to do the right casting and pointer arithmetic to get the 32-bit info value out. Maybe I'm overthinking it.